proposals/stf-general-investment.md

```diff
@@ -0,0 +1,267 @@
+- We currently don’t know how to self-host large repositories such as Nixpkgs, and we’re stuck to GitHub
+
+## What are possible alternatives to your project and how does your project compare to them? (optional) 300 words
+
+- Guix
+
+  Guix is an early fork of Nix using Guile as a configuration and implementation language, and is the only project comparable to the Nix ecosystem.
+  Guix provides its own software distribution and operating system configuration framework, with free software exclusively.
+  The community is much smaller, the package set is rather focused on scientific applications.
+  Guix pioneered full-source bootstrapping, which is now ported to Nixpkgs.
+
+- Arch Linux
+
+  As a software distribution, it has many similarities to Nixpkgs and NixOS in terms of size, structure, and procedures.
+  It does not feature some of the key properties of the Nix ecosystem, such as declarative, composable configurations through uniform functional interfaces.
+  The Arch community has advanced the state of the art in [reproducible builds](https://reproducible-builds.org) in the past, and inspired similar work in Nixpkgs.
+
+## What do you plan to implement with the support from STF? 900 words
+
+*Please describe your objectives and the corresponding activities.
+State clearly how they contribute to the improvement or maintenance of the technology, especially with regard to security and resilience.*
+
+We understand resilience as the ability to recover from disruptions and to adapt to a changing environment.
+Apart from implementing technical measures to that end, merely having qualified engineers available for ongoing maintenance and addressing small-scale, long-term, or cross-cutting issues as part of their engagement will improve the overall health of the ecosystem by unblocking volunteer efforts more quickly than is currently possible.
+We expect this to have a compounding effect on overall productivity, as we estimate that more than 95% of the work is currently done by volunteers, many of whom are slowed down by lack of support and swift decision making.
+
+- Improve the contributor experience (300k)
+
+  Streamlining the contribution process should make it more attractive for newcomers to participate in development, and keep existing people around and motivated.
+  Growing the contributor community will make the ecosystem more robust against fluctuation.
+  We want to achieve this by making more systemic knowledge explicit and easier to discover, access, and manipulate, and thus ease onboarding of new contributors, and reduce contribution turnaround time.
+
+  - Tools for review automation (50k)
+  - Improve code organisation in C++ Nix (30k)
+  - Increase coverage and reduce duration of Nix integration testing (30k)
+  - Monitoring and data gathering for informed decision-making (30k)
+  - Professionalise the moderation team (50k)
+  - Organisation-wide declarative permission and notification management (60k)
```

Comment:
While the "C++ Nix testing" and the "declarative permission management" items include some of it, we don't have an item directly addressing our excessive dependence on GitHub and the scalability and portability challenges we're facing with that. There's probably some potential to collaborate with GitLab or Forejo on enabling support for Nixpkgs-scale self-hosting.